pettingzoo/mpe/scenarios/group_spread.py

```diff
@@ -48,44 +47,37 @@ def make_world(self, groups, num_colors=None, reward_per_group=True, randomise_a
 
     def reset_world(self, world, np_random):
         if self.randomise_all_colors:
-            # generate colors if they are randomised for each episode
-            self.colors = [np.random.random(3) for _ in self.groups]
-            color_idxs = np.arange(len(self.groups))
-        else:
-            # assign one of the available colors to a group
-            color_idxs = np.random.choice(self.num_colors, len(self.groups), replace=False)
-
-        # random properties for agents
+            # TODO: do nicer to avoid overlap here?
+            # randomly generate new colors for each episode
+            self.colors = [np.random.random(3) for _ in range(self.num_colors)]
+
+        # assign one of the available colors to a group
+        color_idxs = np.random.choice(self.num_colors, len(self.groups), replace=False)
```

Comment:
Colors indices should always be picked for each group with indices being between (0, `self.num_colors - 1`). Previously, when randomised colors are used, these were between (0, `len(self.groups) - 1`) which is not the same if `num_colors` argument is provided to the environment! E.g. for `num_colors = 3` when there are only 2 groups, this would have avoided to use the 3rd color ever if randomised colors are used. This probably wouldn't have any meaningful effect since colors in that case are randomised anyway, but seems like odd logic and as far as I can tell there is no reason to make this different depending on whether randomised colors are used?

Reply:
See comment above - I don't think it makes sense to use num_colours with randomise_all_colors, they are conflicting. If randomised colours are used then the colour is select from all possible colours at the start of episode.

Reply:
Now simplified to just get new color values in the random case (handling of how these are selected all happens in the color config definition.